src/community/json-ld/src/main/java/org/geoserver/jsonld/request/JsonLdTemplateCallBackOGC.java

```diff
@@ -0,0 +1,99 @@
+/* (c) 2019 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.jsonld.request;
+
+import org.geoserver.catalog.Catalog;
+import org.geoserver.catalog.FeatureTypeInfo;
+import org.geoserver.config.GeoServer;
+import org.geoserver.jsonld.builders.impl.RootBuilder;
+import org.geoserver.jsonld.configuration.JsonLdConfiguration;
+import org.geoserver.jsonld.expressions.ExpressionsUtils;
+import org.geoserver.jsonld.response.JSONLDGetFeatureResponse;
+import org.geoserver.ows.AbstractDispatcherCallback;
+import org.geoserver.ows.DispatcherCallback;
+import org.geoserver.ows.Request;
+import org.geoserver.platform.Operation;
+import org.geoserver.platform.ServiceException;
+import org.geoserver.util.XCQL;
+import org.geotools.filter.text.ecql.ECQL;
+import org.opengis.filter.Filter;
+import org.springframework.http.HttpHeaders;
+
+/**
+ * This {@link DispatcherCallback} implementation OGCAPI compliant that checks on operation
+ * dispatched event if a json-ld path has been provided to cql_filter and evaluate it against the
+ * {@link org.geoserver.jsonld.builders.JsonBuilder} tree to get the corresponding {@link Filter}
+ */
+public class JsonLdTemplateCallBackOGC extends AbstractDispatcherCallback {
+
+    private Catalog catalog;
+
+    private JsonLdConfiguration configuration;
+
+    public JsonLdTemplateCallBackOGC(GeoServer gs, JsonLdConfiguration configuration) {
+        this.catalog = gs.getCatalog();
+        this.configuration = configuration;
+    }
+
+    @Override
+    public Operation operationDispatched(Request request, Operation operation) {
+        if ("FEATURES".equalsIgnoreCase(request.getService()) && isJsonLdMIME(request)) {
+            try {
+                String filter = (String) request.getKvp().get("FILTER");
+                replaceJsonLdPathWithFilter(filter, operation);
+                String filterLang = (String) request.getKvp().get("FILTER-LANG");
+                replaceJsonLdPathWithFilter(filterLang, operation);
+            } catch (Exception e) {
+                throw new RuntimeException(e);
+            }
+        }
+        return super.operationDispatched(request, operation);
+    }
+
+    private FeatureTypeInfo getFeatureType(String collectionId) {
+        FeatureTypeInfo featureType = catalog.getFeatureTypeByName(collectionId);
+        if (featureType == null) {
+            throw new ServiceException(
+                    "Unknown collection " + collectionId,
+                    ServiceException.INVALID_PARAMETER_VALUE,
+                    "collectionId");
+        }
+        return featureType;
+    }
+
+    private boolean isJsonLdMIME(Request request) {
+        String accept = request.getHttpRequest().getHeader(HttpHeaders.ACCEPT);
+        if (accept != null && accept.contains(JSONLDGetFeatureResponse.MIME)) return true;
```

Comment:
Upside down, "f" override accepts, so it should be tested first.

Reply:
fixed